src/classes/ISObjectRepository.cls

```diff
@@ -4,29 +4,23 @@
 *************************************************************************************************/
 public interface ISObjectRepository {
 
+    // Manipulate queries without executing them
+    ISObjectRepository filterBy(IQueryFilter queryFilter);
+    ISObjectRepository filterBy(List<IQueryFilter> queryFilters);
+
     // SOQL
     SObject getById(Id recordId);
     List<SObject> getById(List<Id> recordIdList);
-    List<SObject> get(List<QueryPredicate> queryPredicates);
-    List<SObject> get(QueryPredicate queryPredicate);
+    List<SObject> get(IQueryFilter queryFilter);
+    List<SObject> get(List<IQueryFilter> queryFilters);
     List<SObject> getByFieldForIds(Schema.SObjectField field, Object value, List<Id> idList);
     List<SObject> getByFieldForIds(Schema.SObjectField field, Object value, Set<Id> idSet);
-    List<SObject> getCreatedSinceTimeValue(Object value);
+    List<SObject> getCreatedSince(Object dateOrDatetime);
```

Comment:
I think removing this method is worth talking about.  The rest of the methods aren't nearly as assumptive, and you can easily accomplish the same thing with the regular .get method.  Thoughts?